src/main/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepository.java

```diff
@@ -77,7 +79,27 @@ public SimpleReactiveCouchbaseRepository(final CouchbaseEntityInformation<T, Str
 	@SuppressWarnings("unchecked")
 	public <S extends T> Mono<S> save(final S entity) {
 		Assert.notNull(entity, "Entity must not be null!");
-		return (Mono<S>) operations.upsertById(entityInformation.getJavaType()).one(entity);
+		// if entity has non-null version property, then replace()
+		if (hasVersionProperty(entity)) {
+			return (Mono<S>) operations.replaceById(entityInformation.getJavaType()).one(entity);
+		} else {
+			return (Mono<S>) operations.upsertById(entityInformation.getJavaType()).one(entity);
+
+		}
+	}
+
+	private boolean hasVersionProperty(Object entity) {
```

Comment:
It can be done. I'll put it in a new Utils class.
I had thought of that, but  SimpleReactiveCouchbaseRepository has a ReactiveCouchbaseOperations, while SimpleCouchbaseRepository has a CouchbaseOperations.   But it doesn't actually need those - it just needs the mapping conext of the converter held by each *Operations - which is the same type.